### DISCUSSION TITLE
Save files won't write to carts for 3V systems

### DISCUSSION THREAD

**Discussion by ameturpoet** [sanni/cartreader#835] [Need help]
I'm having trouble getting save files to write to cartridges for 3V systems (N64 & GBA specifically). When the writing process finishes, it either fails or passes verification depending on how similar the save file was to the one on the cartridge, but it never actually takes effect on the cartridge - the save file there is always unchanged. What could be the issue here?

I can also confirm that the save files created when dumping them seem to be empty, as well. However, rom dumps work just fine - it's just the saves that seem to fail. And the wire supplying the 3V seems securely soldered in place, too.

**Comment by sanni** [comment]
Do you have the special screw drivers needed to open up the cartridges and post a picture of the PCBs? 
Just to see if I have the exact same save chip somewhere to rule out a general issue with that particular save chip.

**Reply by ameturpoet** [reply]
I have done more testing and confirmed that this is actually an issue specifically with EEPROM N64 games. After a few tweaks, everything else works fine, but I have confirmed that those cartridges specifically cannot read/write saves. I uploaded an example save from my Wave Race 64 cartridge, and when I load it up in an emulator, it reads it as a blank save file.

I have so far tested Wave Race 64, Kirby 64, Perfect Dark, Conker, and Yoshi's Story by dumping the save file, making a change in the save on my console, and then trying to write it back. The OSCR always says the write was successful, but doesn't always pass verification. If the change was small, it will say it was successful, but if the change was large, it will fail verification. Regardless, it causes no change to the save on the cartridge.

[WAVE RACE 64.eep.zip](https://github.com/sanni/cartreader/files/12264671/WAVE.RACE.64.eep.zip)

**Reply by sanni** [reply]
That's not a save game, that's the first 64 bytes of the maskrom. It's what the function [idCart()](https://github.com/sanni/cartreader/blob/master/Cart_Reader/N64.ino#L2143C16-L2143C16) read earlier in the code.
This means that the whole eeprom read function completely failed and never read anything out of the eeprom.

Can you flash an earlier version like [V10.2](https://github.com/sanni/cartreader/releases/tag/V10.2) to see how it behaves?

**Reply by ameturpoet** [reply]
I tried 10.2 as well as the latest build, and it didn't seem to affect the output :(

**Reply by sanni** [reply]
Ok, I desoldered the eeprom save chip from one of my cartridges and I get the same result when dumping the now non-existent save with builds later than 10.2. (With 10.2 it just hangs itself up during dumping).

That means that there could be a connection issue on your Cart Reader somewhere. The eeprom is only connected with a couple of lines: eeprom data, eeprom clock, reset. 

You need to check those solder points.

![vtcrtcxrxcrxexer](https://github.com/sanni/cartreader/assets/19169494/95191abd-5df7-428e-83e3-2d821a017d65)

**Comment by Ancyker** [comment]
They sent me the reader to look at since the seller noped out. The person used an SMS-style slot (2.54mm pitch) instead of the N64 slot (2.5mm pitch):

![Image showing an N64 slot, top and the installed slot, bottom](https://github.com/sanni/cartreader/assets/7105333/fccd8e74-d828-43ab-bc0f-87a1b1b37589)

This is causing the pins to bridge:

![Image showing A14 and 15 are bridged to ground](https://github.com/sanni/cartreader/assets/7105333/2c8f1fae-afa9-4e4c-a875-0533345fb5ca)

A14 and 15 here are showing a dead short to each other and ground (the other probe is inserted in a screw hole for ground). If anyone reading this thinking maybe it was an accident, doing this requires going in and changing the footprint of the N64 slot to the same as the one for the SMS slot. The slot won't fit otherwise as the pins won't line up. It was a willful action to use the wrong slot. Because they changed the footprint I can't even fix it without using an entirely different (and correct) board.

In addition, for a bit of a laugh, here is the bulk capacitor for the main PCB compared to one of my own boards to show the "normal" size:

![Image showing that the Chinese seller's capacitor is physically much smaller](https://github.com/sanni/cartreader/assets/7105333/df8217d8-1f39-4000-a34f-8bed703948ba)

Finally, for those lurking here: Based on this, I cannot and do not recommend buying from Aliexpress or Chinese eBay sellers. I believe it's the same seller on both right now (same images/listing, at least), but I'm not sure. It may feel the same as ordering the parts from there, but since there's no economical way to return anything if/when there is an issue, that just means you are out the full value that you paid instead of just a single faulty or incorrect component. With my first order, the seller sent me an SMS slot instead of the N64 slot. I was out $2 as Aliexpress sided with the seller and wanted me to pay to return it.

(Full disclosure: I do sell kits and readers myself, but I'd happily prop up other sellers who actually care about what they are building, such as Bonzo. This is just horrible build quality and isn't really usable.)

**Reply by Hynes84** [reply]
I've purchased a v3 from Aliexpress and am having the same issue. 2.54 is stamped directly on the "N64" slot. Won't read or write eeprom saves. Am I screwed?

**Reply by Hynes84** [reply]
Update: Reverted to firmware V5.5 and was able to write eeprom successfully the "old" way (3.3V, EEP only)
